### PR TITLE
Broken config deadlock test added

### PR DESCRIPTION
**Pull request by olmisa**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

The deadlock test added.

## Related PRs or issues

https://github.com/microsoft/appcenter-sdk-dotnet/issues/1303
https://github.com/microsoft/appcenter-sdk-dotnet/pull/1364

### CODE REVIEW

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 1]
```diff
@@ -830,6 +830,20 @@ public void HandleUserConfirmationAsyncAlwaysSendWhileDisabled()
             var alwaysSendValue = AppCenter.Instance.ApplicationSettings.GetValue(Crashes.PrefKeyAlwaysSend, false);
             Assert.IsFalse(alwaysSendValue);
         }
+        
+        [TestMethod]
+        [Timeout(5000)]
+        public void BrokenConfigDeadlockTest()
+        {
+            var settings = new DefaultApplicationSettings();
+            var expectedLines = System.IO.File.ReadAllLines(DefaultApplicationSettings.FilePath);
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs:839]: Add using for `System.IO`

Comment (olmisa) [reply Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs]: Fixed

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 2]
```diff
@@ -830,6 +831,20 @@ public void HandleUserConfirmationAsyncAlwaysSendWhileDisabled()
...
             Assert.IsFalse(alwaysSendValue);
         }
+        
+        [TestMethod]
+        [Timeout(5000)]
+        public void BrokenConfigDeadlockTest()
+        {
+            var settings = new DefaultApplicationSettings();
+            var expectedLines = SystemFile.ReadAllLines(DefaultApplicationSettings.FilePath);
+            var corruptedLines = new string[expectedLines.Length - 2];
```
Comment (annakocheshkova) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs:841]: Explanation why "2"?

Comment (olmisa) [reply Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs:846]: I don't think that it is very important. Just remove several lines from 'XML' to make it corrupted.

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 3]
```diff
@@ -831,6 +832,29 @@ public void HandleUserConfirmationAsyncAlwaysSendWhileDisabled()
             Assert.IsFalse(alwaysSendValue);
         }
 
+        /// <summary>
+        /// Test that after corrupted config file restored
+        /// the App Center start could not be locked (for example, locked by call AppCenter.SetEnabledAsync(false)).
```
Comment (Jamminroot) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs:837]:
```suggestion
        /// Test that after restoring corrupted config doesn't lock App Center start routine.
        /// (for example, locked by call AppCenter.SetEnabledAsync(false)).
```

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 4]
```diff
@@ -831,6 +832,29 @@ public void HandleUserConfirmationAsyncAlwaysSendWhileDisabled()
...
+            var corruptedLines = new string[expectedLines.Length - 2];
+            Array.Copy(expectedLines, 0, corruptedLines, 0, corruptedLines.Length);
+            SystemFile.WriteAllLines(DefaultApplicationSettings.FilePath, corruptedLines);
+
+            // Config file restored.
+            settings = new DefaultApplicationSettings();
+
+            // Lock situation reproduction.
+            AppCenter.Start("appSecret", typeof(Crashes));
+            AppCenter.SetEnabledAsync(false);
```
Comment (Jamminroot) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs:855]: I think we can add a comment about implicit verification through test timeout.

**Pull request merged**